telethon/network/connection/tcpintermediate.py

```diff
@@ -8,12 +10,47 @@ class ConnectionTcpIntermediate(Connection):
     Intermediate mode between `ConnectionTcpFull` and `ConnectionTcpAbridged`.
     Always sends 4 extra bytes for the packet length.
     """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._codec = IntermediatePacket()
+
     def _init_conn(self):
-        self._writer.write(b'\xee\xee\xee\xee')
+        self._writer.write(self._codec.tag)
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself) (as I said above, with this change, it makes sense to factor out all these steps).

Reply:
yep, agree